addon_imps/tests/storage/test_gitlab.py

```diff
@@ -0,0 +1,148 @@
+        self.network = AsyncMock(spec_set=HttpRequestor)
+        self.imp = GitlabStorageImp(config=self.config, network=self.network)
+
+    def _patch_get(self, return_value: dict | list, status=200):
+        mock = self.network.GET.return_value.__aenter__.return_value
+        mock.json_content = AsyncMock(return_value=return_value)
+        mock.http_status = status
+        mock.headers = {}
+
+    def _assert_get(self, url: str, query: dict = None):
+        extra_params = {"query": query} if query else {}
+        self.network.GET.assert_called_once_with(url, **extra_params)
+        self.network.GET.return_value.__aenter__.assert_awaited_once()
+        self.network.GET.return_value.__aenter__.return_value.json_content.assert_awaited_once()
+        self.network.GET.return_value.__aexit__.assert_awaited_once_with(
+            None, None, None
+        )
+
+    async def test_get_external_account_id(self):
+        mock_response = {"user_id": "12345"}
+        self._patch_get(mock_response)
+
+        result = await self.imp.get_external_account_id({})
+        self.assertEqual(result, "12345")
+        self._assert_get("user/preferences")
+
+    async def test_list_root_items(self):
+        mock_response = [
+            {
+                "id": "1",
+                "name": "repo1",
+                "path_with_namespace": "repo1",
+            },
+            {
+                "id": "2",
+                "name": "repo2",
+                "path_with_namespace": "repo2",
+            },
+        ]
+        self._patch_get(mock_response)
+
+        result = await self.imp.list_root_items()
+        expected_items = [
+            ItemResult(item_id="repo1:", item_name="repo1", item_type=ItemType.FOLDER),
+            ItemResult(item_id="repo2:", item_name="repo2", item_type=ItemType.FOLDER),
+        ]
+        expected_result = ItemSampleResult(items=expected_items)
+
+        self.assertEqual(result.items, expected_result.items)
+        self._assert_get(
+            "projects",
+            {
+                "membership": "true",
+                "simple": "true",
+                "pagination": "true",
+                "sort": "asc",
+            },
+        )
+
+    async def test_get_item_info_repository(self):
+        mock_response = {
+            "id": "1",
+            "name": "repo1",
+            "path_with_namespace": "repo1",
+        }
+        self._patch_get(mock_response)
+        result = await self.imp.get_item_info("1:")
+
+        expected_result = ItemResult(
+            item_id="repo1:", item_name="repo1", item_type=ItemType.FOLDER
+        )
+        self.assertEqual(result, expected_result)
+        self._assert_get("projects/1")
+
+    async def test_get_item_info_file(self):
+        file_mock = {
+            "file_name": "README.md",
+            "path": "README.md",
+        }
+
+        self._patch_get(file_mock)
+        self.network.GET.return_value.__aenter__.return_value.headers = {}
```

Comment:
I'd suggest to implement this in `patch_get`?
But moreover, is this important to specify empty response headers here?

Reply:
Actually, it is not necessary